### DISCUSSION TITLE
Como solucionar la autenticación y permiso denegado

### DISCUSSION THREAD

**Discussion by Solcalderon** [community/community#52769] [Repositories] [answered]
### Select Topic Area

Question

### Body

Como solucionar cuando intento subir un archivo a un repositorio y me aparece permiso denegado o falla de autenticación ya he visto muchos tutoriales y ninguno me funciona

**Comment by Dhoony21** [comment]
Fgy

**Comment by sadeqi-ah** [comment, marked as answer]
If you are having trouble uploading a file to a GitHub repository and receiving a “permission denied” or “authentication failed” message, there are several things you can try:

1. Check that you are logged into your GitHub account and have permission to access the repository.
2. If you are using SSH to connect to the repository, make sure you have added your SSH key correctly to your GitHub account.
	
To add an SSH key to your GitHub account, follow these steps:

1. Generate a new SSH key pair on your local machine by running the command ﻿`ssh-keygen -t ed25519 -C "your_email"` in your terminal or command prompt. This will create a public and private key pair. 
2. Copy the public key by running the command ﻿`cat ~/.ssh/id_rsa.pub` and copying the output.
3. Log in to your GitHub account and go to your account settings.
4. Click on “SSH and GPG keys” in the left sidebar.
5. Click on “New SSH key” and give your key a title.
6. Paste the public key you copied earlier into the “Key” field.
7. Click “Add SSH key” to save the key to your account.

You should now be able to use your SSH key to authenticate with your GitHub account when cloning or pushing to repositories.

**Comment by ghostinhershell** [comment]
Hi @Solcalderon, thanks for participating!

We take our [Code of Conduct](https://github.com/community/community/blob/0ca37cf594e438fd4aec3c7f3736b95e6b0b3567/CODE_OF_CONDUCT.md) very seriously and want to help ensure that everyone has a good experience free of antagonism and harassment. Unfortunately, we don’t currently have moderators for languages other than English. Until that changes, we need to ask that everyone use English here in the GitHub Community when posting. We’ll be locking any posts in languages other than English for now, including this one.
